arangod/VocBase/Methods/Collections.h

```diff
@@ -120,9 +122,10 @@ struct Collections {
       double timeout                      // single-server drop timeout
   );
 
-  static Result warmup(TRI_vocbase_t& vocbase, LogicalCollection const& coll);
+  static futures::Future<Result> warmup(TRI_vocbase_t& vocbase,
+                                        LogicalCollection const& coll);
 
-  static Result revisionId(Context& ctxt, TRI_voc_rid_t& rid);
+  static futures::Future<OperationResult> revisionId(Context& ctxt);
```

Comment:
perhaps `ResultT` would be more appropriate here ? But idk

Reply:
Simple enough to just use the VPack in this case, and then we don't need to deal with `boost::optional`.